modules/SatelliteController/SatelliteController.cpp

```diff
@@ -24,8 +27,11 @@ void SatelliteController::init() {
     invoke_init(*p_evse_manager);
     invoke_init(*p_system);
 
-    this->rpc = std::make_unique<rpc::client>(this->config.hostname, this->config.port);
-
+    //
+    // register all callbacks for our desired interfaces
```

Comment:
Does it work then to move this subscribtion block after the do..while loop? If not, should we add checks to aviod the crashing?

Reply:
The point is that I try to avoid such boilerplate code as much as possible - per definition, the pointer has to be set and must be valid and if this is not the case, then I want the code to crash so that the whole system restarts. This is why I added the description above the block. (BTW: I'll update the description - the english is poor and puzzling)